### PR TITLE
V2 translator mtls

### PR DESCRIPTION
**Pull request by ilackarms**
sets mtls configuration in the translator by configuring destination rules and the `meshPolicy` crd

### CODE REVIEW

File: pkg/translator/istio/translator.go [hunk 1]
```diff
@@ -84,27 +85,100 @@ type labelsPortTuple struct {
...
+		}
+		rules := routingRulesByMesh[mesh]
+		in := inputMeshConfig{
+			mesh:  mesh,
+			rules: rules,
+		}
+		meshConfig, err := t.translateMesh(params, in, upstreams, resourceErrs)
+		if err != nil {
+			return nil, nil, err
+		}
```
Comment (yuval-k) [inline pkg/translator/istio/translator.go]: this errors all meshes if one of them has errored. is this desired?

Comment (ilackarms) [reply pkg/translator/istio/translator.go]: fixed

File: pkg/translator/istio/translator.go [hunk 2]
```diff
@@ -84,27 +89,107 @@ type labelsPortTuple struct {
...
 	upstreams := snapshot.Upstreams.List()
 	routingRules := snapshot.Routingrules.List()
-	encryptionRules := snapshot.Ecryptionrules.List()
 
 	resourceErrs := make(reporter.ResourceErrors)
 	resourceErrs.Accept(meshes.AsInputResources()...)
 	resourceErrs.Accept(meshGroups.AsInputResources()...)
 	resourceErrs.Accept(routingRules.AsInputResources()...)
-	resourceErrs.Accept(encryptionRules.AsInputResources()...)
+	resourceErrs.Accept(upstreams.AsInputResources()...)
```
Comment (ilackarms) [inline pkg/translator/istio/translator.go]: add a comment that we need to know if gloo is installed here, if so we dont want to validate upstreams ourselves

File: pkg/translator/istio/translator_encryption.go [hunk 3]
```diff
@@ -7,49 +7,28 @@ import (
...
-	"github.com/solo-io/supergloo/pkg/translator/istio/plugins"
-
 	"github.com/solo-io/go-utils/stringutils"
+	"github.com/solo-io/solo-kit/pkg/api/v1/reporter"
 	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
 	"github.com/solo-io/supergloo/pkg/api/external/istio/networking/v1alpha3"
+	"github.com/solo-io/supergloo/pkg/translator/istio/plugins"
 )
 
 func (t *translator) makeDestinatioRuleForHost(
```
Comment (ilackarms) [inline pkg/translator/istio/translator_encryption.go]: merged  makeDestinatioRuleForHost and initDestinationRule

**Pull request closed without merging**